### ISSUE TITLE
The treatment of model inputs in the short-term series forecasting framework (self.model(batch_x, None, dec_inp, None)), and the accuracy of the TimesNet model at multiple inputs and single outputs (MS).

### ISSUE THREAD

**Issue by StorywithLove** [thuml/Time-Series-Library#599]
**Thanks for the great work!**

I experimented with the short-term prediction framework on my own dataset, and the results are as follows. I have two main questions asking for your help.

(1) The way the inputs are processed during model training
`outputs = self.model(batch_x, None, dec_inp, None)`
When using Custom data processing, is it understood that no processed temporal features are input to the model?

(2) Specific effects of the TimesNet.
The information on the own dataset is as follows:
![image](https://github.com/user-attachments/assets/197b6eab-1b1e-456a-841f-26a4e68a59d1)
The hyperparameters are as follows:
```
export CUDA_VISIBLE_DEVICES=0

model_name=TimesNet

python -u run.py \
  --task_name short_term_forecast \
  --is_training 1 \
  --data custom \
  --root_path ./dataset \
  --data_path Haining.csv \
  --seasonal_patterns 'Daily' \
  --freq d \
  --seq_len 14 \
  --label_len 7 \
  --pred_len 7 \
  --model_id HN_daily \
  --model $model_name \
  --features MS \
  --enc_in 5 \
  --dec_in 5 \
  --e_layers 2 \
  --d_layers 1 \
  --factor 3 \
  --c_out 1 \
  --batch_size 16 \
  --d_model 16 \
  --d_ff 16 \
  --top_k 5 \
  --des 'Exp' \
  --itr 1 \
  --learning_rate 0.001 \
  --loss 'SMAPE'
```

The TimesNet training log is as follows:
```
Basic Config
  Task Name:          short_term_forecast Is Training:        1
  Model ID:           HN_daily            Model:              TimesNet

Data Loader
  Data:               custom              Root Path:          ./dataset
  Data Path:          Haining.csv         Features:           MS
  Target:             OT                  Freq:               d
  Checkpoints:        ./checkpoints/      

Forecasting Task
  Seq Len:            14                  Label Len:          7
  Pred Len:           7                   Seasonal Patterns:  Daily
  Inverse:            1

Model Parameters
  Top k:              5                   Num Kernels:        6
  Enc In:             5                   Dec In:             5
  C Out:              1                   d model:            16
  n heads:            8                   e layers:           2
  d layers:           1                   d FF:               16
  Moving Avg:         25                  Factor:             3
  Distil:             1                   Dropout:            0.1
  Embed:              timeF               Activation:         gelu

Run Parameters
  Num Workers:        0                   Itr:                1
  Train Epochs:       10                  Batch Size:         16
  Patience:           3                   Learning Rate:      0.001
  Des:                Exp                 Loss:               SMAPE
  Lradj:              type1               Use Amp:            0

GPU
  Use GPU:            1                   GPU:                0
  Use Multi GPU:      0                   Devices:            0,1,2,3

De-stationary Projector Params
  P Hidden Dims:      128, 128            P Hidden Layers:    2

Use GPU: cuda:0
>>>>>>>start training : short_term_forecast_HN_daily_TimesNet_custom_ftMS_sl14_ll7_pl7_dm16_nh8_el2_dl1_df16_expand2_dc4_fc3_ebtimeF_dtTrue_Exp_0>>>>>>>>>>>>>>>>>>>>>>>>>>
train 852
val 120
Epoch: 1 cost time: 21.77793574333191
Epoch: 1, Steps: 54 | Train Loss: 88.6896116 Vali Loss: 20.0594559 Test Loss: 20.0594559
Validation loss decreased (inf --> 20.059456).  Saving model ...
Updating learning rate to 0.001
Epoch: 2 cost time: 1.6076772212982178
Epoch: 2, Steps: 54 | Train Loss: 83.8424207 Vali Loss: 19.3244629 Test Loss: 19.3244629
Validation loss decreased (20.059456 --> 19.324463).  Saving model ...
Updating learning rate to 0.0005
Epoch: 3 cost time: 1.5542120933532715
Epoch: 3, Steps: 54 | Train Loss: 82.5396876 Vali Loss: 19.3047676 Test Loss: 19.3047676
Validation loss decreased (19.324463 --> 19.304768).  Saving model ...
Updating learning rate to 0.00025
Epoch: 4 cost time: 1.5465478897094727
Epoch: 4, Steps: 54 | Train Loss: 80.1366471 Vali Loss: 18.8032112 Test Loss: 18.8032112
Validation loss decreased (19.304768 --> 18.803211).  Saving model ...
Updating learning rate to 0.000125
Epoch: 5 cost time: 1.5525996685028076
Epoch: 5, Steps: 54 | Train Loss: 80.0868904 Vali Loss: 18.8944645 Test Loss: 18.8944645
EarlyStopping counter: 1 out of 3
Updating learning rate to 6.25e-05
Epoch: 6 cost time: 1.5358200073242188
Epoch: 6, Steps: 54 | Train Loss: 79.6192151 Vali Loss: 19.0623188 Test Loss: 19.0623188
EarlyStopping counter: 2 out of 3
Updating learning rate to 3.125e-05
Epoch: 7 cost time: 1.5397529602050781
Epoch: 7, Steps: 54 | Train Loss: 80.2205770 Vali Loss: 18.8357239 Test Loss: 18.8357239
EarlyStopping counter: 3 out of 3
Early stopping
>>>>>>>testing : short_term_forecast_HN_daily_TimesNet_custom_ftMS_sl14_ll7_pl7_dm16_nh8_el2_dl1_df16_expand2_dc4_fc3_ebtimeF_dtTrue_Exp_0<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
test 243
test shape: (243, 7, 1)
mse:54095.71484375, mae:147.4913787841797, rmse:232.58485412597656, mape:0.08083653450012207

```

The test accuracies for each model are as follows:
**I would like to ask if there is a more appropriate hyperparameter configuration to tune the TimesNet model for single output multiple input.**
**Thanks again for the great work!**

![image](https://github.com/user-attachments/assets/59ccbb4f-8a8a-405b-8522-d32c369d7613)
